packages/components/src/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,385 @@
+                  h: parseInt(hours, 10),
+                  m: parseInt(minutes, 10),
+                })
+              : null
+          );
+          onFocusChange(!focus);
+        }}
+      >
+        <Text mt={0} mb={2} bold>
+          Uhrzeit wählen
+        </Text>
+        <Grid mb={3}>
+          <GridItem width={[1, 1, 2 / 3]}>
+            <Grid>
+              <GridItem width="64px" mr={1}>
+                <Input
+                  name="hour"
+                  value={hours}
+                  type="text"
+                  placeholder="hh"
+                  onChange={(e) =>
+                    e.currentTarget.value.match(/\b(^$|[0-9]|1[0-9]|2[0-3])\b/)
+                      ? setHours(e.currentTarget.value)
+                      : !e.currentTarget.value
+                      ? setHours('')
+                      : null
+                  }
+                  maxLength={2}
+                  autoComplete="off"
+                  hideReset
+                />
+              </GridItem>
+              <GridItem
+                display="flex"
+                alignItems="center"
+                justifyContent="center"
+                width="auto"
+                px={0}
+              >
+                <Text my={0}>:</Text>
+              </GridItem>
+              <GridItem width="64px" mx={1}>
+                <Input
+                  name="minute"
+                  value={minutes}
+                  type="text"
+                  placeholder="mm"
+                  onChange={(e) =>
+                    e.currentTarget.value.match(/\b(^$|[0-9]|[12345][0-9])\b/)
+                      ? setMinutes(e.currentTarget.value)
+                      : !e.currentTarget.value
+                      ? setMinutes('')
+                      : null
+                  }
+                  maxLength={2}
+                  autoComplete="off"
+                  hideReset
+                />
+              </GridItem>
+              <GridItem
+                display="flex"
+                alignItems="center"
+                justifyContent="flex-start"
+                width="auto"
+                px={0}
+              >
+                <Text my={0}>Uhr</Text>
+              </GridItem>
+            </Grid>
+          </GridItem>
+          <GridItem
+            width={[1, 1, 1 / 3]}
+            display="flex"
+            justifyContent="flex-end"
+          >
+            <Button mt={[2, 2, 0]} width={[1, 1, 'auto']} type="submit">
+              Datum festlegen
+            </Button>
+          </GridItem>
+        </Grid>
+      </form>
+    </Box>
+  );
+};
+
+export const DatePicker: React.FC<{
+  id: string;
+  timePicker?: boolean;
+  date: moment.Moment | null;
+  onChange: (date: moment.Moment | null) => void;
+}> = ({ id, timePicker = false, date, onChange }) => {
+  const [focus, setFocus] = useState(false);
+  const [mobile, setMobile] = useState(false);
+  const [hours, setHours] = useState('');
```

Comment:
Hours und Minutes bitte nicht in einem eigenen State halten, das lässt sich hier vermeiden indem man es einfach immer aus dem von oben hereingegebenen Datum nimmt. Dann hat der Timepicker auch weniger Logik und braucht im onChange nur die ausgewählte Zeit zum Datum hinzufügen und nach oben weiter geben.